### ISSUE TITLE
Images cropped to (-1px, -1px)

### ISSUE THREAD

**Issue by drewoldag** [lincc-frameworks/fibad#127]
Working with the HSC dataset with images that vary generally by 1 or 2 pixels at most. The expected behavior is that we'll identify the smallest image and crop all other images to that size. However, we're identifying -1x-1 as the smallest size and attempting to crop to that size, which produces downstream problems.

Explicitly setting `crop_to = [X, X]` is a work around, but likely requires that the user know the minimum size before hand. 

> [2024-12-03 15:19:00,488 fibad:INFO] Runtime Config read from: ../../drew_config.toml
[2024-12-03 15:19:00,585 fibad.data_sets.hsc_data_set:INFO] Processed 1000 objects for pruning
[2024-12-03 15:19:00,586 fibad.data_sets.hsc_data_set:INFO] Pruned 0 out of 1000 objects
[2024-12-03 15:19:00,587 fibad.data_sets.hsc_data_set:INFO] Checking file dimensions to determine standard cutout size...
[2024-12-03 15:19:00,591 fibad.data_sets.hsc_data_set:WARNING] Some images differ from the mean width or height of all images by more than 1px
Images will be cropped to (-1px, -1px)
See data/hsc_8asec_1000/Attempted for an example image of width -1px
See data/hsc_8asec_1000/Attempted for an example image of height -1px
[2024-12-03 15:19:00,592 fibad.data_sets.hsc_data_set:INFO] HSC Data set loader has 1000 objects
[2024-12-03 15:19:00,593 fibad.data_sets.hsc_data_set:INFO] Test split contains 0 items
[2024-12-03 15:19:00,594 fibad.data_sets.hsc_data_set:INFO] Train split contains 800 items
[2024-12-03 15:19:00,595 fibad.data_sets.hsc_data_set:INFO] Validation split contains 200 items
/home/drew/miniconda3/envs/fibad/lib/python3.12/site-packages/torch/nn/init.py:453: UserWarning: Initializing zero-element tensors is a no-op
  warnings.warn("Initializing zero-element tensors is a no-op")
2024-12-03 15:19:00,612 ignite.distributed.auto.auto_dataloader INFO: Use data loader kwargs for dataset '<fibad.data_sets.hsc': 
	{'sampler': <torch.utils.data.sampler.SubsetRandomSampler object at 0x7f8a4cf12a80>, 'batch_size': 64, 'shuffle': False, 'num_workers': 2, 'pin_memory': True}
2024-12-03 15:19:00,614 ignite.distributed.auto.auto_dataloader INFO: Use data loader kwargs for dataset '<fibad.data_sets.hsc': 
	{'sampler': <torch.utils.data.sampler.SubsetRandomSampler object at 0x7f8a4cf10740>, 'batch_size': 64, 'shuffle': False, 'num_workers': 2, 'pin_memory': True}
[2024-12-03 15:19:00,623 fibad.pytorch_ignite:INFO] Training model on device: cuda
[2024-12-03 15:19:00,624 fibad.pytorch_ignite:INFO] Total epochs: 20
Current run is terminating due to exception: Calculated padded input size per channel: (2 x 2). Kernel size: (3 x 3). Kernel size can't be greater than actual input size
Engine run is terminating due to exception: Calculated padded input size per channel: (2 x 2). Kernel size: (3 x 3). Kernel size can't be greater than actual input size

**Comment by drewoldag** [comment]
The message indicates: "See .../Attempted for an example image of width -1px". Should we just filter out "Attempted" files that are found in the manifest?

**Comment by drewoldag** [comment]
```
filters_ref = sorted(filters_ref)
        self.prune_count = 0
        for index, (object_id, filters) in enumerate(self.files.items()):

            # ! Adding the following check _seems_ to address the issue

            # Drop objects that failed to download
            if any("Attempted" in v for v in filters.items()):
                msg = f"Attempted to download {object_id} but failed. Pruning."
                self._mark_for_prune(object_id, msg)
                break

            # ! End of additional check

            # Drop objects with missing filters
            filters = sorted(list(filters))
            if filters != filters_ref:
                msg = f"HSCDataSet in {self.path} has the wrong group of filters for object {object_id}."
                self._mark_for_prune(object_id, msg)
                logger.info(f"Filters for object {object_id} were {filters}")
                logger.debug(f"Reference filters were {filters_ref}")
```

**Comment by mtauraso** [comment]
Thanks, I thought about this a bit more and I think the right place to do this is before pruning, which I'm addressing in PR#130  

My heuristic here is that unreliable data should be remove as soon as possible, and should not affect other aspects of the analysis.  Dropping bad images in the prune stage hazards the situation that a single filter image being bad (which may not be relevant to the current analysis) would cause all filters for an object to be dropped. This seems like counter-intuitive behavior. 

Correcting our earlier in-person conversation: The download code is *correct* to report a (-1px,-1px) dimension for an attempted download, because a value for the manifest data structure is required there. The key observation is that when the download code writes this value into memory, it must maintain the property that the in-memory data structure is a valid on-disk manifest because the write of the manifest occurs on the main thread, and downloading occurs on worker threads.  

In HSCDataSet I've added an error that occurs on read-in for the case where we end up automatically

**Comment by aritraghsh09** [comment]
So, I have a situation where images are being cropped to [0,0], not due to some issue with loading or how the minimum size is being calculated, but because in the (download) manifest file, one entry that is a successful download has a size = of [0,0] 

[For specifics, this is with our 0.25 < z < 0.50 HSC data; and this specific issue is with Object ID -- 43342890100872869]

A physics limitation on cutout size is > 0; is it possible to add this in the pruning step without adding too much overhead (in the situation where the manifest file already has all the sizes)? 

P.S: My current workaround has been to supply a filter catalog that doesn't include that specific object-id
